cmd/validate-domains/main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/projectdiscovery/public-bugbounty-programs/pkg/core"
+	"github.com/tidwall/gjson"
+)
+
+var bbListFile = flag.String("file", "../../chaos-bugbounty-list.json", "Chaos bugbounty list json file")
+
+func main() {
+	flag.Parse()
+
+	rawJSON, err := os.ReadFile(*bbListFile)
+	if err != nil {
+		log.Fatalf("Failed to read initial JSON file: %v", err)
+	}
+
+	var ps core.ChaosList
+	err = json.Unmarshal(rawJSON, &ps)
+	if err != nil {
+		log.Fatalf("Failed to parse initial JSON file: %v", err)
+	}
+
+	var invalidDomains []string
+	gdata := gjson.ParseBytes(rawJSON)
+	gdata.Get("programs.#.domains|@flatten").ForEach(func(key, value gjson.Result) bool {
+		domain := value.String()
```

Comment:
The empty value would be considered valid with this check as `core.ExtractHostname("")` returns `""`. Consider implementing tabular-tests to ensure that most basic edge cases are covered by the logic. An incomplete list can be:
- [ ] Empty value (`""`)
- [ ] Spaces (`" "`)
- [ ] contains wildcard ("*")
- [ ] URL (`"http://`, `https://`)
- [ ] non primary/secondary dns name (`xxx.projectdiscovery.io`)
- [ ] ...